PHPfiles/ProductCalendar.php

```diff
@@ -54,6 +69,147 @@ function renderCalendar() {
         HtmlHelpers::writeBreakLine();
         HtmlHelpers::writeBreakLine();        
     }
-}
 
+    private static function renderNoLunchTableCell($productDate) {
+        $dateString = date_format(new DateTime($productDate), "M d, Y");
+        HtmlHelpers::writeTableCell("$dateString <p><i>No lunch</i></p>", ThemeConstants::TableCellNothingToOrderStyle);
+    }
+
+    private static function renderNoProductDetailsTableCell() {
+        HtmlHelpers::writeTableCell("&nbsp;", ThemeConstants::TableCellNothingToOrderStyle);                
+    }
+
+    private static function updateProductStateForTiming($productId, $productDetails, $orderLateDateTime, $orderTooLateDateTime) {
+        $productTimingKey = get_post_meta($productId, 'timing_key', true);
+
+        if ($productTimingKey == ProductOrderTiming::TooLate) {
+            // nothing to do, product already marked as too late
+            return ProductOrderTiming::TooLate;
+        }
+
+        if ($productTimingKey == ProductOrderTiming::KindaLate) {
+            // check if it's too late to order
+            if (time() > $orderTooLateDateTime) {
+                update_post_meta($productId, 'timing_key', ProductOrderTiming::TooLate);
+                return ProductOrderTiming::TooLate;
+            }
+            return ProductOrderTiming::KindaLate;
+        } 
+        
+        // product currently marked ontime, check if it's kindda late, or too late
+        if (time() < $orderLateDateTime) {
+            return ProductOrderTiming::OnTime;
+        } elseif (time() < $orderTooLateDateTime) {
+            // update product timing state
+            update_post_meta($productId, 'timing_key', ProductOrderTiming::KindaLate);
+
+            // update product's inventory count
+            if ($productDetails->get_stock_quantity() > 50) {
+                wc_update_product_stock($productId, 50);
+            }
+
+            //Get and set new price for each variation of the parent product
+			$mealVariationIds = $productDetails->get_visible_children();
+            
+            foreach ($mealVariationIds as $currentVariationId) {
+                $productVariation = wc_get_product($currentVariationId);
+                $RegPrice = $productVariation->get_regular_price();
+                $lateprice = $RegPrice + 1;
+                update_post_meta($currentVariationId, '_regular_price', $lateprice);
```

Comment:
when i was reading today about updating cart and order meta data, one example was adding meta associated with an additional price to the cart.  in the future, i may consider doing that method instead of altering the product regular price.  so for example, it would add a note that says "late order" to the cart item and that note have a $1 charge that gets added to the cart, as part of the product being added.  Just FYI.

Reply:
Sounds good.